proposals/field-keyword-nullability.md

```diff
@@ -0,0 +1,88 @@
+# `field` keyword nullability
+
+## Goals
+
+Ensure a reasonable level of null-safety for various usage patterns of the `field` keyword feature. Avoid requiring the user to apply `[field: AllowNull, MaybeNull]` or similar attributes to the property in order to express common patterns around lazy initialization, etc.
+
+## Terms
+
+A property is a *field-backed property* if it meets any of the following conditions:
+1. It contains any automatically implemented accessors (`get;`/`set;`/`init;`).
+2. It uses the `field` keyword within any accessor bodies.
+
+Some examples of *field-backed properties* include:
+```cs
+public string Prop1 { get; set; }
+public string Prop2 { get => field; set; }
+public string? Prop3 { get => field; }
+```
+
+The variable denoted by the `field` keyword in a property's accessors is the *backing field* of that property.
+
+## Nullability of the *backing field*
+
+The *backing field* has the same type as the property. However, its nullable annotation may differ from the property. To determine this nullable annotation, we introduce the concept of *null-resiliency*. *Null-resiliency* intuitively means that the property's `get` accessor behaves properly even when the field contains the `default` value for its type.
```

Comment:
Consider spelling out "properly" in plain language: ie, does not return a maybe-null value, and has no nullable warnings.

Reply:
I feel the language in the next paragraph accomplishes this

> Then, if nullable analysis of the getter yields no nullable warnings, the property is *null-resilient*.

Reply:
I would be fine with either deleting the "behaves properly" sentence or changing it to more closely resemble the quoted sentence.

Reply:
I have addressed this by saying:

> *Null-resiliency* intuitively means that the property's `get` accessor preserves null-safety even when the field contains the `default` value for its type.